sympy/ntheory/tests/test_totient_hypothesis.py

```diff
@@ -0,0 +1,15 @@
+from hypothesis import given
+from hypothesis import strategies as st
+from sympy.ntheory import totient
+from sympy import divisors
+import random
+
+
+@given(n=st.integers(1,10**10))
+def test_totient_hypothesis(n):
+    assert totient(n) <= n
+    div = divisors(n)
+    totients = [totient(i) for i in div]
+    assert n == sum(totients)
+    d = random.choice(div)
+    assert totient(n // (n // d)) == totient(d)
```

Comment:
I'm actually unclear just what the point of this test is. If `d` is a divisor of `n`, shouldn't `n // (n // d)` be equal to `d`? Maybe a test for `divisors()` could check that `all(n % d == 0 for d in div)`, but otherwise, assuming that is true, I don't see what this is checking about `totient`.